iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageInstallationScriptExecutionCheck.java

```diff
@@ -0,0 +1,155 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Argument;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+import static org.sonar.iac.docker.symbols.ArgumentResolution.Status.UNRESOLVED;
+
+@Rule(key = "S6505")
+public class PackageInstallationScriptExecutionCheck implements IacCheck {
+
+  private static final String MESSAGE = "Omitting --ignore-scripts can lead to the execution of shell scripts. Make sure it is safe here.";
+
+  private static final String REQUIRED_FLAG = "--ignore-scripts";
+
+  private enum StatementType {
+    NPM,
+    YARN
+  }
+
+  private static final Map<String, StatementType> TRIGGER_COMMANDS_STATEMENT_TYPE_MAPPING = Map.of(
+    "npm", StatementType.NPM,
+    "pnpm", StatementType.NPM,
+    "yarn", StatementType.YARN);
+
+  private static final Set<String> NPM_INSTALL_ALIASES = Set.of("ci", "add", "i", "in", "ins", "inst", "insta", "instal", "isnt", "isnta", "isntal", "isntall");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, PackageInstallationScriptExecutionCheck::checkRunInstruction);
+  }
+
+  private static void checkRunInstruction(CheckContext ctx, RunInstruction instruction) {
+    StatementValidator validator = new StatementValidator(ctx, instruction.arguments());
+    validator.processArguments();
+  }
+
+  private static class StatementValidator {
+
+    CheckContext ctx;
+    List<Argument> arguments;
+    PackageManagerStatement currentStatement;
+
+    StatementValidator(CheckContext ctx, List<Argument> arguments) {
+      this.ctx = ctx;
+      this.arguments = arguments;
+    }
+
+    void processArguments() {
+      if (arguments == null || arguments.isEmpty()) {
```

Comment:
The arguments field of the RunInstruction can not be null but the list can be empty.